### PR TITLE
[Import Group] Merge triples cache

### PR DESCRIPTION
**Pull request by shifucun**
Due to the nested proto structure, it's not possible to use a protobuf for response and make it backward compatible. A lot of the change has been on data converting in this PR.

The diff from current and import group mode matches reasonably. This changes added some fields that are missing in the current cache, like "subjectTypes" "subjectName" etc.

Applied sorting to both old and new golden files so they can be compared

### CODE REVIEW

File: internal/server/handler.go [hunk 1]
```diff
@@ -219,12 +220,13 @@ func (s *Server) GetPropertyLabels(
 		return nil, err
 	}
 	jsonRaw, err := protojson.Marshal(resp)
-	// To make the API response backward compatible. This is to add the outer
-	// level `{\"data\":`
-	jsonRaw = jsonRaw[8 : len(jsonRaw)-1]
 	if err != nil {
 		return nil, err
 	}
+	// To make the API response backward compatible. This is to add the outer
```
Comment (Spaceenter) [inline internal/server/handler.go:227]: s/add/remove ?

File: internal/server/node/convert.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	"sort"
+
+	pb "github.com/datacommonsorg/mixer/internal/proto"
+	"github.com/datacommonsorg/mixer/internal/server/model"
+)
+
+// Convert pb.GetTriplesResponse to legacy golang struct to make the API response
+// backward compatible.
+func ToLegacyResult(in *pb.GetTriplesResponse) map[string][]*model.Triple {
+	out := map[string][]*model.Triple{}
```
Comment (Spaceenter) [inline internal/server/node/convert.go]: nit: "out" might confuse the readers that this means "outEntity", so maybe change the variable name to "result"?

File: internal/server/node/golden/get_triples_test.go [hunk 3]
```diff
@@ -31,107 +31,115 @@ import (
...
+			}
+			resp, err := client.GetTriples(ctx, req)
+			if err != nil {
+				t.Errorf("could not GetTriples: %s", err)
+				continue
+			}
+			var result map[string][]*model.Triple
+			err = json.Unmarshal([]byte(resp.GetPayload()), &result)
+			if err != nil {
+				t.Errorf("Can not Unmarshal payload")
```
Comment (Spaceenter) [inline internal/server/node/golden/get_triples_test.go:110]: t.Errorf("Can not Unmarshal payload: %s", err)

File: internal/server/node/golden/get_triples_test.go [hunk 4]
```diff
@@ -31,107 +31,115 @@ import (
...
+			if c.limit > 0 {
+				req.Limit = c.limit
+			}
+			resp, err := client.GetTriples(ctx, req)
+			if err != nil {
+				t.Errorf("could not GetTriples: %s", err)
+				continue
+			}
+			var result map[string][]*model.Triple
+			err = json.Unmarshal([]byte(resp.GetPayload()), &result)
```
Comment (Spaceenter) [inline internal/server/node/golden/get_triples_test.go:108]: if err := json.Unmarshal(...); err != nil {

**Review by Spaceenter — APPROVED** [review]

File: internal/server/handler.go [hunk 5]
```diff
@@ -219,12 +221,13 @@ func (s *Server) GetPropertyLabels(
...
 	jsonRaw, err := protojson.Marshal(resp)
-	// To make the API response backward compatible. This is to add the outer
-	// level `{\"data\":`
-	jsonRaw = jsonRaw[8 : len(jsonRaw)-1]
 	if err != nil {
 		return nil, err
 	}
+	// To make the API response backward compatible. This is to remove the outer
+	// level `{\"data\":`
+	jsonRaw = jsonRaw[8 : len(jsonRaw)-1]
```
Comment (pradh) [inline internal/server/handler.go:229]: nit: just wondering, do we not need to skip the trailing "}" ?

**Review by pradh — APPROVED** [review]

File: internal/server/handler.go (hunk 5 above)
Comment (shifucun) [reply internal/server/handler.go:229]: Oh, it's doing that. Added in the comments

File: internal/server/node/convert.go (hunk 2 above)
Comment (shifucun) [reply internal/server/node/convert.go]: Done

**Comment by shifucun** [comment]
Thanks for review!

**Pull request merged**